Extract file detail

phileducncc opened the pull request.

phileducncc committed:
feat: Add Extract Detail component for file and URL data extraction

- Implemented ExtractDetailComponent with file upload and URL input options.
- Added loading spinner and error handling for file selection.
- Created a responsive table to display extracted product data.
- Integrated branch selection and validation for extracted data.
- Added functionality to save extracted data to the server.
- Enhanced UI with CSS styles for better user experience.
- Updated NormalDetailTableComponent to include Extract functionality.
- Added API endpoint for bulk creation of extracted details.

phileducncc committed:
feat: Enhance table layout and input styles for better usability

phileducncc committed:
feat: Refactor extract detail component and add total calculation for extracted data

tiennguyenhuuncc commented on aspnet-core/src/FinanceManagement.Application/APIs/OutcomingEntryDetails/OutcomingEntryDetailAppService.cs:
```diff
@@ -62,6 +62,58 @@ public async Task<GetOutcomingEntryDetailDto> Create(GetOutcomingEntryDetailDto
         }
+        
+        [HttpPost]
+        [AbpAuthorize(PermissionNames.Finance_OutcomingEntry_OutcomingEntryDetail_TabDetailInfo_Create)]
```
```
public async Task CreateMany([FromBody] List<GetOutcomingEntryDetailDto> inputList)
        {
            if (inputList == null || !inputList.Any())
            {
                throw new UserFriendlyException("Danh sách dữ liệu trống.");
            }

            var outcomingEntryId = inputList.First().OutcomingEntryId;

            var outcomingEntry = await WorkScope.GetAll<OutcomingEntry>()
                .Include(s => s.WorkflowStatus)
                .Include(s => s.OutcomingEntryDetails)
                .Where(s => s.Id == outcomingEntryId)
                .FirstOrDefaultAsync();

            if (outcomingEntry == null)
            {
                throw new UserFriendlyException($"Không tìm thấy Request chi Id {outcomingEntryId}");
            }

            if (outcomingEntry.WorkflowStatus?.Code?.Trim() != FinanceManagementConsts.WORKFLOW_STATUS_START)
            {
                throw new UserFriendlyException($"Chỉ có thể tạo mới thông tin chi tiết khi Request chi có mã trạng thái {FinanceManagementConsts.WORKFLOW_STATUS_START}");
            }

            foreach (var input in inputList)
            {               
                var entity = ObjectMapper.Map<OutcomingEntryDetail>(input);
                entity.OutcomingEntryId = outcomingEntryId;
                await WorkScope.InsertAsync(entity);
            }

            await CurrentUnitOfWork.SaveChangesAsync();

            outcomingEntry.Value = outcomingEntry.OutcomingEntryDetails.Sum(x => x.Total);
            await WorkScope.UpdateAsync(outcomingEntry);
                        
        }
```

phileducncc committed:
Refactor and enhance expenditure request detail components

- Updated import statements for consistency in app.module.ts.
- Improved error handling and success messages in ExtractDetailComponent.
- Adjusted layout and button configurations in normal-detail-table.component.html.
- Refined logic and formatting in normal-detail-table.component.ts for better readability and performance.
- Streamlined request-detail.service.ts methods for clarity and consistency.
- Modified OutcomingEntryDetailAppService.cs to improve exception messages and remove unnecessary return values.
